core/ssl-only/slack-invites/class-edd-slack-app-invites-settings.php

```diff
@@ -113,80 +111,91 @@ public function add_slack_invites_settings( $oauth_settings ) {
 					),
 					'multiple' => true,
 					'chosen' => true,
-					'options' => $this->get_public_channels(),
+					'options' => $public_channels,
 					'placeholder' => sprintf( _x( 'Just #%s', 'Just #general Channel Invite', 'edd-slack' ), $this->general_channel ),
 					'std' => array(),
 					'desc' => sprintf( _x( 'The <code>#%s</code> Channel is always granted by default. Choose any other additional Channels you would like to auto-invite %s to.', 'Channels for Vendors Description Text', 'edd-slack' ), $this->general_channel, EDD_FES()->helper->get_vendor_constant_name( true, true ) )
 				);
 
 			}
-			
+
 		}
-		
+
 		$oauth_settings = array_merge( $oauth_settings, $slack_invites_settings );
-		
+
 		return $oauth_settings;
-		
+
 	}
-	
+
 	/**
 	 * Returns all Public Slack Channels from the Slack API
-	 * 
-	 * @access		public
-	 * @since		1.1.0
-	 * @return		array Slack Channels
+	 *
+	 * @access  public
+	 * @since   1.1.0
+	 * @return  array Slack Channels
 	 */
 	public function get_public_channels() {
-		
+
+		$channels_array = array();
 		// Don't bother if we aren't granting Client Scope
-		if ( ! edd_get_option( 'slack_app_has_client_scope', false ) ) return array();
-		
-		if ( ! $channels_array = maybe_unserialize( get_transient( 'edd_slack_channels_list' ) ) ) {
-		
-			$result = EDDSLACK()->slack_api->get( 'channels.list' );
+		if ( ! edd_get_option( 'slack_app_has_client_scope', false ) ) {
+			return $channels_array;
+		}
+		if ( defined( 'DOING_AJAX' ) && DOING_AJAX ) {
+			return $channels_array;
+		}
 
-			$channels = $result->channels;
+		$transient = maybe_unserialize( get_transient( 'edd_slack_channels_list' ) );
+		if ( $transient ) {
+			return $transient;
+		}
 
-			$channels_array = array();
-			foreach ( $channels as $channel ) {
+		$oauth_token = edd_get_option( 'slack_app_oauth_token', false );
+		// Don't bother if we don't have an OAUTH Token
+		if ( ! $oauth_token || '-1' == $oauth_token ) {
+			return $channels_array;
+		}
 
-				if ( $channel->is_general ) {
+		$result = EDDSLACK()->slack_api->get( 'conversations.list' );
 
-					// If necessary, update our General Channel
-					$this->general_channel = ( $channel->name !== $this->general_channel ) ? $channel->name : $this->general_channel;
+		if ( empty( $result->channels ) ) {
```

Comment:
```suggestion
		if ( empty( $result->channels ) || ! is_array( $result->channels ) ) {
```

Don't see how that would happen, but just in case since we're about to treat it like one. :shrug: